src/pool.c

```diff
@@ -93,6 +93,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define MAX_BAD_SHARES 5
 #define MAX_DOWNSTREAM 8
 #define MAX_HOST 256
+#define MAX_RIG_ID 32
```

Comment:
Only 32 workers per account basically only caters to small scale miners.

Reply:
MAX_RIG_ID tracks the max length of the rig id that can be specified
MAX_RIGS tracks the maximum number of rigs to keep track of, if 1024 is still too should be able to still increase.

Reply:
Well, you _were_ using `MAX_RIG_ID*MAX_RIG_ID+128` for the buffer hence only 36 workers per account. MAX_RIG_ID is also redefined. 1024 is also too small. Per other comments, this buffer should be about 1 MB and which allows for around 25k workers per account and the function that fills the buffer is simplified by passing start and end buffer pointers.